[com_content] Filter on publish_up and publish_down

n9iels opened the pull request:
### Summary of Changes

This PR adds the possibility to filter article on the "Publish Up" and "Finish Publishing" date.
### Testing Instructions
1. Apply this PR
2. Create some article with different "publish up" and "finish publishing" dates
3. Select the following option in the "Sort table by" field and confirm that the articles are sorted correctly:
   - Start Publishing ascending
   - Start Publishing descending
   - Finish Publishing ascending
   - Finish Publishing descending

n9iels committed:
Filter on publish_up and publish_down

brianteeman commented on administrator/components/com_content/models/forms/filter_articles.xml:
```diff
@@ -106,6 +106,10 @@
 			<option value="association DESC" requires="associations">JASSOCIATIONS_DESC</option>
 			<option value="a.created_by ASC">JAUTHOR_ASC</option>
 			<option value="a.created_by DESC">JAUTHOR_DESC</option>
+			<option value="a.publish_up ASC">COM_CONTENT_PUBLISH_UP_ASC</option>
```
Why did you randomly insert it here?

brianteeman commented:
For me this is going to make it very confusing as the only date that is displayed is the created date. So as soon as people start to use these new sorts they will wonder why it doesnt work as the Date column does not reflect the ordering

n9iels commented:
I think it is not confusing, because `Start Publishing ascending

gwsdesk commented:
I agree with Brian here. If we start manipulating again the ordering system we get riots on the forums and sorry @n9iels I disagree with it. It is not logic but confusing

brianteeman commented:
@n9iels they are different things but if a user doesnt see it as there is a no column I can guarantee they will be confused

@gwsdesk this is nothing to do with front end ordering

n9iels commented:
Hmm true...
What if we changed that column of the filter is on?  So if we filter on start published date the "date" column is replaced by the start published date. Or will that be confusing too?

brianteeman commented:
In my opinion yes

On 24 August 2016 at 09:25, Niels van der Veer <notifications@github.com>
wrote:

> Hmm true...
> What if we changed that column of the filter is on? So if we filter on
> start published date the "date" column is replaced by the start published
> date. Or will that be confusing too?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/joomla/joomla-cms/pull/11670#issuecomment-241992452>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABPH8fbEkGT4cI9AZWg3ZNLOrGdntpTXks5qi__egaJpZM4JojEB>
> .
>



-- 
Brian Teeman
Co-founder Joomla! and OpenSourceMatters Inc.
http://brian.teeman.net/

gwsdesk commented:
I did not say frontend. I thought this is meant for admin ordering so if 
I am mistaken me sorry


On 8/24/2016 3:18 PM, Brian Teeman wrote:
>
> @n9iels <https://github.com/n9iels> they are different things but if a 
> user doesnt see it as there is a no column I can guarantee they will 
> be confused
>
> @gwsdesk <https://github.com/gwsdesk> this is nothing to do with front 
> end ordering
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub 
> <https://github.com/joomla/joomla-cms/pull/11670#issuecomment-241990784>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/AHzLNpniQ9facFehQ4X4qThb11Xm8hwxks5qi_4_gaJpZM4JojEB>.
>

sanderpotjer commented:
For people that are working heavily with scheduled articles it does makes a lot of sense to be able to order the backend articles list on the publication date. They don't even look at the creation date. At the moment there is no possibility to see the (upcoming) articles in the backend ordered by the publication date, while many frontend views are using the publication date as ordering. It would be great to display articles in the same order as on the frontend.

I do think that @n9iels suggestion of changing the "Date" column in the overview based on the filtering selection. So for all people used to the current listing nothing will change. 

As soon as someone selects the "Publishing date" (the suggested new option) in the filter the date column shows the Publishing date instead of the Creation date. I would only recommend to change "Date" column heading into "Date Created" heading, right now it is not clear what date is displayed anyway.

brianteeman commented:
@sanderpotjer you've convinced me - so if the date column changes are added to this PR then it can be tested <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11670">issues.joomla.org/tracker/joomla-cms/11670</a>.</sub>

n9iels committed:
Place the filter after the created filter

n9iels committed:
Change date column when apply the filter

n9iels committed:
Change heading when publish filter is applied

n9iels committed:
Add language string

n9iels commented:
@sanderpotjer @brianteeman I changed the position of the publish_up filter in the select list. When the filter is applied the publish_up dates are visible instead of the created dates. Also the heading is different when the filter is applied

brianteeman commented:
I have tested this item :red_circle: unsuccessfully on 0a3c98fb371ffff2599090ba27074d35ad4a947b<br /><br />PHP Parse error:  syntax error, unexpected ';', expecting ':' in /administrator/components/com_content/views/articles/tmpl/default.php on line 86
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11670">issues.joomla.org/tracker/joomla-cms/11670</a>.</sub>

zero-24 commented on administrator/components/com_content/views/articles/tmpl/default.php:
```diff
@@ -81,7 +81,11 @@
-							<?php echo JHtml::_('searchtools.sort', 'JDATE', 'a.created', $listDirn, $listOrder); ?>
+							<?php if (stripos($listOrder, 'publish') !== false) : ?>
+								<?php echo JHtml::_('searchtools.sort', 'COM_CONTENT_HEADING_DATE_CREATED', 'a.publish_up', $listDirn, $listOrder); ?>
+							<?php else; ?>
```
this needs to be `else :`

n9iels committed:
Fix syntax error

n9iels commented:
@brianteeman @zero-24 I fixed the syntax error.

Offtopic: Why didn't Travis mention this error?

zero-24 commented:
IIRC Travis complete ignore the tmpl folders ;)

brianteeman commented:
I have tested this item :red_circle: unsuccessfully on e5a42f3408f360da32d17d70408e9aa29594c69d<br /><br />1. Set filter to Start Publishing Ascending in the filter and the column header says Date Created

2. Set filter to Finish Publishing Ascending in the filter and the column header says Date Created

3. When the filter is set to Finish Publishing and an item does not have a finish publishing date (most will not) then the value displayed is -0001-11-30<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11670">issues.joomla.org/tracker/joomla-cms/11670</a>.</sub>

mbabker commented:
>> Offtopic: Why didn't Travis mention this error?

> IIRC Travis complete ignore the tmpl folders ;)

They aren't completely ignored, just the code standard checks mostly.  We don't have a check for valid syntax in the files though.

zero-24 commented:
>They aren't completely ignored, just the code standard checks mostly. We don't have a check for valid syntax in the files though.

Thanks for corrcting me. Here is a a RFC fpr PHP Syntax checker https://github.com/joomla/joomla-cms/pull/12379

n9iels committed:
Merge branch 'staging' into filter-publish

euismod2336 committed:
fixes for publish/unpublish

added language strings. now shows '-' instead of invalid date when no publish date. show correct header

euismod2336 committed:
changed language string

start published -> start publishing

csthomas committed:
Do it in simpler way

IMHO this should not be that complicated. Do it simpler.

csthomas committed:
Small fix

csthomas commented:
Please consider my suggestion from https://github.com/n9iels/joomla-cms/pull/5

euismod2336 commented:
Maybe it would be even easier to declare the $orderingColumn at the top and fill it via a regex (or substr), instead of setting it somewhere and using it somewhere else. You could then also skip the strpos and immediately do if (orderingColum == publish_up)

csthomas committed:
Declare the $orderingColumn at the top

csthomas commented:
@euismod2336  As you wish https://github.com/n9iels/joomla-cms/pull/5

euismod2336 committed:
Merge pull request #5 from csthomas/patch-1

Do it in simpler way

euismod2336 committed:
Merge branch 'staging' into filter-publish

csthomas commented:
The same changes should be added to https://github.com/joomla/joomla-cms/blob/staging/administrator/templates/hathor/html/com_content/articles/default.php

euismod2336 committed:
Hathor now also uses 'date created'

euismod2336 committed:
Merged branch filter-publish into filter-publish

euismod2336 commented:
Hathor doesn't provide an option to sort on date, so I only changed the language string so that it's more obvious the creation date is meant in the table.

brianteeman commented:
I have tested this item :white_check_mark: successfully on f3c552bdd79c317b31d3e68ad10ef80e6eb63780<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11670">issues.joomla.org/tracker/joomla-cms/11670</a>.</sub>

csthomas commented:
What about featured articles view? It might be worthwhile to add the same?

euismod2336 committed:
added same functionality to featured articles view

euismod2336 commented:
You are correct, again :) thank you @csthomas

csthomas commented on administrator/components/com_content/models/forms/filter_featured.xml:
```diff
@@ -101,7 +101,11 @@
+            <option value="a.publish_down ASC">COM_CONTENT_PUBLISH_DOWN_ASC</option>
+            <option value="a.publish_down DESC">COM_CONTENT_PUBLISH_DOWN_DESC</option>
+            <option value="language ASC">JGRID_HEADING_LANGUAGE_ASC</option>
 			<option value="language DESC">JGRID_HEADING_LANGUAGE_DESC</option>
```
Please correct indentation.

csthomas commented:
There is also JDATE at https://github.com/joomla/joomla-cms/blob/staging/administrator/templates/hathor/html/com_content/featured/default.php#L108

csthomas committed:
Update filter_featured.xml and template hathor featured/default.php

euismod2336 committed:
Merge pull request #6 from csthomas/patch-1

Update filter_featured.xml and template hathor featured/default.php

sanderpotjer commented:
@euismod2336 can you please fix the code conflict in `administrator/components/com_content/views/articles/tmpl/default.php` with the latest changes? Right now it is removing the latest changes around hits, votes & rating:
<img width="352" alt="screen shot 2016-11-04 at 14 00 55" src="https://cloud.githubusercontent.com/assets/522834/20006295/2ada94bc-a297-11e6-8f0c-527ef6ac6d5b.png">

euismod2336 committed:
Merge branch 'staging'

Conflicts:
	administrator/components/com_content/views/articles/tmpl/default.php

euismod2336 commented:
something seems terribly broken at this point...one moment :)

euismod2336 commented:
ignore previous comment, that was related to missing language files in my local copy. @sanderpotjer @marclight (i got your name from the related issue) could you give this a test run? Then maybe we can get this done before 2020 ;)

infograf768 commented on administrator/language/en-GB/en-GB.com_content.ini:
```diff
@@ -185,6 +188,10 @@ COM_CONTENT_URL_FIELD_B_BROWSERNAV_LABEL="URL B Target Window"
 COM_CONTENT_URL_FIELD_C_BROWSERNAV_LABEL="URL C Target Window"
 COM_CONTENT_WARNING_PROVIDE_VALID_NAME="Please provide a valid, non-blank title."
 COM_CONTENT_XML_DESCRIPTION="Article management component."
+COM_CONTENT_PUBLISH_UP_ASC="Start Publishing ascending"
```
Please alpha order these new strings.

euismod2336 committed:
moved/reorderd new strings

infograf768 commented:
I have tested this item :white_check_mark: successfully on 5c43460d6426e671beb034e5701153b50e716100<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11670">issues.joomla.org/tracker/joomla-cms/11670</a>.</sub>

sanderpotjer commented:
I have tested this item :white_check_mark: successfully on 5c43460d6426e671beb034e5701153b50e716100<br /><br />Thanks for this feature! <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11670">issues.joomla.org/tracker/joomla-cms/11670</a>.</sub>

jeckodevelopment commented:
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11670">issues.joomla.org/tracker/joomla-cms/11670</a>.</sub>
